app/src/main/java/com/yacgroup/yacguide/utils/SearchBarHandler.kt

```diff
@@ -52,8 +53,10 @@ class SearchBarHandler(searchBarLayout: ConstraintLayout,
             })
         }
 
-        searchBarLayout.findViewById<EditText>(R.id.searchEditText).apply {
+        searchBarLayout.findViewById<TextInputLayout>(R.id.searchTextInputLayout).apply {
             setHint(searchHintResource)
+        }
+        searchBarLayout.findViewById<EditText>(R.id.searchEditText).apply {
```

Comment:
You have changed the widget's type in the xml to `com.google.android.material.textfield.TextInputEditText`. Does this still correspond to Kotlin's `EditText` type?

Reply:
It's a sub-class of `EditText` (https://developer.android.com/reference/com/google/android/material/textfield/TextInputEditText) which shall be used within the `TextInputLayout`.